src/test/unit/lib/cache/cache.test.js

```diff
@@ -0,0 +1,153 @@
+/**************************************************************************
+ *  (C) Copyright ModusBox Inc. 2019 - All rights reserved.               *
+ *                                                                        *
+ *  This file is made available under the terms of the license agreement  *
+ *  specified in the corresponding source code repository.                *
+ *                                                                        *
+ *  ORIGINAL AUTHOR:                                                      *
+ *       James Bush - james.bush@modusbox.com                             *
+ **************************************************************************/
+
+'use strict';
+
+
+jest.mock('redis');
+
+const Cache = require('@internal/cache');
+jest.unmock('@internal/cache');
+
+
+const defaultConfig = {
+    host: 'dummyhost',
+    port: 1234,
+    logger: console
+};
+
+
+const dummyPubMessage = {
+    data: 12345,
+    test: '98765'
+};
+
+
+const createCache = async() => {
+    const cache = new Cache(defaultConfig);
+    await cache.connect();
+    return cache;
+};
+
+
+describe('Cache', () => {
+
+    test('Makes connections to redis server for cache operations', async () => {
+        const cache = await createCache();
+        expect(cache).not.toBeFalsy();
+
+        // the cache should have opened a general connection
+        expect(cache.client).not.toBeFalsy();
+
+        // the cache should have opened a subscription connection
+        expect(cache.subscriptionClient).not.toBeFalsy();
+    });
+
+
+    test('Makes subscriber callbacks on the correct channels when messages arrive', async () => {
+        const cache = await createCache();
+        const msg1 = JSON.parse(JSON.stringify(dummyPubMessage));
```

Comment:
Maybe better to put this deep copy into `beforeEach`, and rename `const dummyPubMessage = { ... }` definition to `dummyPubMessageTemplate`

Reply:
done